### ISSUE TITLE
Using async - await with Express router

### ISSUE THREAD

**Issue by LuisEnMarroquin** [standard/eslint-config-standard-with-typescript#613]
**What version of this package are you using?**

```shell
$ npx ts-standard --version
10.0.0
```

**What operating system, Node.js, and npm version?**

```shell
$ uname && node -v && npm -v
MINGW64_NT-10.0-19043
v14.17.3
6.14.13
```

**What happened?**

When using async-await with Express everything works as expected but I get an error when linting:

```shell
$ npm run lint

> @ lint C:\Users\USERNAME\Repos\express-ts-standard
> ts-standard --fix

ts-standard: Typescript Standard Style! (https://github.com/standard/ts-standard)
  C:\Users\USERNAME\Repos\express-ts-standard\src\router.ts:5:24: Promise returned in function argument where a void return was expected. (@typescript-eslint/no-misused-promises)
```

Here is my code for the route, I need to use async because I do a lot of request to other apis.
And I dont want to do a lot of nested `.then`

```ts
import { Request, Response } from 'express'
import axios from 'axios'

export const doRequest = async (_: Request, res: Response): Promise<Response> => {
  try {
    const fullURL = 'https://jsonplaceholder.typicode.com/todos/1'
    const requestResult = await axios.get(fullURL)
    return res.status(200).json(requestResult.data)
  } catch (error) {
    return res.status(500).json(error)
  }
}
```

Here a small repo as example: https://github.com/LuisEnMarroquin/express-ts-standard

I know I can use `// eslint-disable-line @typescript-eslint/no-misused-promises`
But is probably not good to populate all the code with that comment

**What did you expect to happen?**

I just want to avoid this warning

```
Promise returned in function argument where a void return was expected. (@typescript-eslint/no-misused-promises)
```

Or find a way to properly use async - await according to this rule

**Are you willing to submit a pull request to fix this bug?**

If it's an easy PR yes

**Comment by mightyiam** [comment]
I'm a bit confused by the line number `5` that is reported. What does that pertain to?

**Comment by LuisEnMarroquin** [comment]
@mightyiam operating system, Node.js, and npm version

MINGW64_NT-10.0-19043 === Windows 10 build 19043

**Comment by mightyiam** [comment]
@LuisEnMarroquin in this output:

```
  C:\Users\USERNAME\Repos\express-ts-standard\src\router.ts:5:24: Promise returned in function argument where a void return was expected. (@typescript-eslint/no-misused-promises)
```

The number 5 refers to line 5 of that file. But that seems to be

```
  try {
```

Are you certain that this file you pasted is the one that was checked?

**Comment by LuisEnMarroquin** [comment]
That is the function that is imported in `router.ts` and used with `router.get`, here is the code:

```shell
import { doRequest } from './controller'
import { Router } from 'express'
const router = Router()

router.get('/example', doRequest)

export default router
```

[Here](https://github.com/LuisEnMarroquin/express-ts-standard/tree/main/src) is a small project with 3 files

![image](https://user-images.githubusercontent.com/35376824/125474741-89b0a222-5ffe-4348-ab7c-cd28e21ee931.png)

**Comment by LinusU** [comment]
I think the problem is that you have declared your function to return `Promise<Response>`, which makes the linter (and users reading the code I think as well) think that express is trying to use the return value of the function. But since the express function is declared as taking a void-returning function the linter believes that you have missed something.

In this specific case, I would recommend either using the Express.js 5.x beta which allows you to do this without the `try`/`catch` since it accepts async functions and will catch the error. Or use a small shim that wraps the function for you, e.g:

```typescript
type Method = 'get' | 'post' | 'put' | 'delete'
type Handler = (req: Request) => Promise<void>
function route (method: Method, path: string, handler: Handler): void {
  router[method](path, (req, res, next) => { handler(req).catch(next) })
}
```

```diff
- router.get('/example', doRequest)
+ route('get', '/example', doRequest)
```

**Comment by LuisEnMarroquin** [comment]
Hi @LinusU `Promise<Response>` it's because TS forces me to return a promise, I can use `Promise<void>` and the problem will persist

The code that you sent me kind of works, but it adds some hard to read code. 
And I will have to make the following funcion (that would make code much harder to read) and add a validation is the Handler is `Promise<void>` of `void`, so I'm able to reuse it also with not `async` funcions

```js
const router = Router()
type Method = 'get' | 'post' | 'put' | 'delete'
type HandlerPromise = (req: Request, res: Response) => Promise<Response>
export function route (method: Method, path: string, handler: HandlerPromise): void {
  router[method](path, (req: Request, res: Response, next: NextFunction) => { handler(req, res).catch(next) })
}
export const routes = router
```

Is there a way I can override that rule on `.eslintrc` or `package.json`, tried with this but wont work:

```json
"eslintConfig": {
    "rules": {
      "@typescript-eslint/no-misused-promises": 0
    }
},
```

Otherwise I'll have to populate almost every route with: `// eslint-disable-line @typescript-eslint/no-misused-promises`

![image](https://user-images.githubusercontent.com/35376824/129824967-980fbeaa-ca54-4bba-968f-4366425ef3b1.png)

About `express v5` I won't be confortable using it because is still in beta and could cause me some issues later

**Comment by itsritiksingh** [comment]
hi @LuisEnMarroquin  , have solved this issue because i'm getting this same problem but i couldn't find any solution.

**Comment by LuisEnMarroquin** [comment]
@itsritiksingh Aparently the only solution is adding this comment `// eslint-disable-line @typescript-eslint/no-misused-promises` or using express V5

**Comment by rodgerhe** [comment]
@LuisEnMarroquin @itsritiksingh 

Try `express-async-handler`:
```
import asyncHandler from 'express-async-handler'

export const doRequest = async (_: Request, res: Response) => {
  try {
    const fullURL = 'https://jsonplaceholder.typicode.com/todos/1'
    const requestResult = await axios.get(fullURL)
    res.status(200).json(requestResult.data)
  } catch (error) {
    res.status(500).json(error)
  }
}

router.get('/test', asyncHandler(doRequest))
```

U can also create your own utility wrapper:
```
const asyncWrapper = (asyncFn: (req: Request, res: Response) => Promise<void>) => {
  function (req: Request, res: Response, next: NextFunction) {
    asyncFn(req, res).catch(next)
  }
}
```

Also you can use type assertion, which is a bit hacky:
```
import { RequestHandler } from 'express'

router.get('/example', doRequest as unknown as RequestHandler)
```
